source/geometry_model/spherical_shell.cc

```diff
@@ -372,6 +372,47 @@ namespace aspect
 
 
 
+    template <int dim>
+    void
+    SphericalShell<dim>::adjust_positions_for_periodicity (Point<dim> &position,
+                                                           ArrayView<Point<dim>> connected_positions) const
+    {
+      AssertThrow(dim == 2,
+                  ExcMessage("Periodic boundaries currently "
+                             "only work with 2D spherical shell."));
+      AssertThrow(phi == 90,
+                  ExcMessage("Periodic boundaries currently "
+                             "only work with 90 degree opening angle in spherical shell."));
+
+      if (periodic)
+        {
+          // define a rotation matrix for the new position depending on the boundary
+          Tensor<2,dim> rotation_matrix;
+
+          if (position[0] < 0.)
+            {
+              rotation_matrix[0][1] = 1.;
+              rotation_matrix[1][0] = -1.;
+            }
+          else if (position[1] < 0.)
+            {
+              rotation_matrix[0][1] = -1.;
+              rotation_matrix[1][0] = 1.;
+            }
+          else
+            return;
```

Comment:
You can probably make this easier to understand if you used
```suggestion
          Tensor<2,dim> rotation_matrix;

          if (position[0] < 0.)
            {
              rotation_matrix = Physics::Transformations::Rotations::rotation_matrix_2d(...plus or minus pi/2);
            }
          else if (position[1] < 0.)
            {
              rotation_matrix = Physics::Transformations::Rotations::rotation_matrix_2d(...minus or plus pi/2);
            }
          else
            return;
```
But then I realized that the function only exists in 2d and you have dimension independent code here. So forget about it.